velox/exec/tests/MultiFragmentTest.cpp

```diff
@@ -935,3 +935,30 @@ TEST_F(MultiFragmentTest, exchangeDestruction) {
   leafTask = nullptr;
   rootTask = nullptr;
 }
+
+TEST_F(MultiFragmentTest, cancelledExchange) {
+  // Create a source fragment borrow the output type from it.
+  auto planFragment = exec::test::PlanBuilder()
+                          .tableScan(rowType_)
+                          .filter("c0 % 5 = 1")
+                          .partitionedOutput({}, 1, {"c0", "c1"})
+                          .planFragment();
+
+  // Create task with exchange.
+  auto planFragmentWithExchange =
+      exec::test::PlanBuilder()
+          .exchange(planFragment.planNode->outputType())
+          .partitionedOutput({}, 1)
+          .planFragment();
+  auto exchangeTask =
```

Comment:
How we ensure the race condition can always happen like ExchangeClient::next get calls before the task is aborted? Thanks!

Reply:
I don't know how to ensure the race always happens.
Usually it is either not easy or not possible without a lot of custom code.
In our case running "buck test --stress-runs 100" was guaranteed to produce at least two failures. The race is subtle.

Reply:
We can use testvalue to enforce it to be triggered.